mlflow/models/evaluation/default_evaluator.py

```diff
@@ -540,6 +614,21 @@ def plot_pr_curve():
 
         self._log_pandas_df_artifact(per_class_metrics_collection_df, "per_class_metrics")
 
+    def _evaluate_custom_metrics(self):
+        if self.custom_metrics is None:
+            return
+        copy_of_y_pred = copy.deepcopy(self.y_pred)
+        copy_of_y = copy.deepcopy(self.y)
+        copy_of_metrics = copy.deepcopy(self.metrics)
+        for i, custom_metric in enumerate(self.custom_metrics):
+            # recreating the dataframe for each custom metric function call,
+            # in case the user modifies the eval_df inside their custom function.
+            eval_df = pd.DataFrame({"prediction": copy_of_y_pred, "target": copy_of_y})
```

Comment:
Will this step create another copy from copy_of_y_pred and copy_of_y (and a copy of dict is temporarily created)? Why not just moving this line outside of the loop?

Reply:
Actually I just remembered that Corey brought up a good point [here](https://github.com/mlflow/mlflow/pull/5389#discussion_r813242477) that each custom metric function could modify `eval_df`. So it might be a good idea to perform deep copy on each. Namely reverting back to:
```python
eval_df = {"prediction": copy.deepcopy(self.y_pred), "target": copy.deepcopy(self.y)}
```